File: app/src/main/java/com/kickstarter/ui/viewholders/EmptyProfileViewHolder.java
package com.kickstarter.ui.viewholders;

import android.view.View;
import android.widget.Button;

import com.kickstarter.R;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import butterknife.Bind;
import butterknife.ButterKnife;
import butterknife.OnClick;

public final class EmptyProfileViewHolder extends KSViewHolder {
  protected @Bind(R.id.explore_projects_button) Button exploreButton;

  private final Delegate delegate;

  public interface Delegate {
    void emptyProfileViewHolderExploreProjectsClicked(EmptyProfileViewHolder viewHolder);
  }

  public EmptyProfileViewHolder(final @NonNull View view, final @NonNull Delegate delegate) {
    super(view);
    this.delegate = delegate;
    ButterKnife.bind(this, view);
  }

  @Override
  public void bindData(final @Nullable Object data) throws Exception {}

  @Override
  public void onBind() {}

  @OnClick(R.id.explore_projects_button)
  public void exploreProjectsClicked() {
    this.delegate.emptyProfileViewHolderExploreProjectsClicked(this);
  }
}

File: app/src/main/java/com/kickstarter/ui/viewholders/EmptyProfileViewHolder.kt
package com.kickstarter.ui.viewholders

import com.kickstarter.databinding.ProfileEmptyStateViewBinding

class EmptyProfileViewHolder(
    private val binding: ProfileEmptyStateViewBinding,
    private val delegate: Delegate
) : KSViewHolder(binding.root) {

    interface Delegate {
        fun emptyProfileViewHolderExploreProjectsClicked(viewHolder: EmptyProfileViewHolder)
    }

    @Throws(Exception::class)
    override fun bindData(data: Any?) {
    }

    override fun onBind() {
        binding.exploreProjectsButton.setOnClickListener {
            exploreProjectsClicked()
        }
    }

    private fun exploreProjectsClicked() {
        delegate.emptyProfileViewHolderExploreProjectsClicked(this)
    }
}
